usaspending_api/common/csv_helpers.py

```diff
@@ -5,31 +5,31 @@
 from usaspending_api.common.retrieve_file_from_uri import RetrieveFileFromUri
 
 
-def count_rows_in_csv_file(filename, has_header=True, safe=True):
+def count_rows_in_csv_file(filename, has_header=True, safe=True, delimiter=","):
     """
-        Simple and efficient utility function to provide the rows in a vald CSV file
+        Simple and efficient utility function to provide the rows in a vald delimited file
         If a header is not present, set head_header parameter to False
 
-        Added "safe" mode which will handle any NUL BYTE characters in CSV files
+        Added "safe" mode which will handle any NUL BYTE characters in delimited files
         It does increase the function runtime by approx 10%.
             Example:
-                "non-safe" counting ~1 million records in a CSV takes 9s
+                "non-safe" counting ~1 million records in a delimited file takes 9s
                 using "safe mode", it now takes 10s
 
     """
     with codecs.open(filename, "r") as f:
         if safe:
-            row_count = sum(1 for row in csv.reader((line.replace("\0", "") for line in f)))
+            row_count = sum(1 for row in csv.reader((line.replace("\0", "") for line in f), delimiter=delimiter))
         else:
-            row_count = sum(1 for row in csv.reader(f))
+            row_count = sum(1 for row in csv.reader(f, delimiter=delimiter))
     if has_header and row_count > 0:
         row_count -= 1
 
     return row_count
 
 
 # Function inspired by Ben Welsh: https://gist.github.com/palewire/596056
```

Comment:
Do we really need inspiration comments?

Reply:
Did you look at the linked code before making this comment?

Reply:
The code is almost a copy from the link in the comment. I believe the original dev who brought this code into the codebase did a straight-copy. It has since been improved with better variable names, and Python3 syntax.

This comment is a "weak attribution" to a snippet of code in the ether of the internet. Is it Open Source? Are we allowed to use it? Should it be removed?

At minimum, a link to the original author is provided to enlighten future devs and the public.